From 5bdf7b42580cfb2744a56f088737422a10685033 Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Tue, 19 May 2020 17:10:27 +0200
Subject: [PATCH 21/24] config_distro_bootcmd: Add platform start script hook
 for networking

This can be used by boards to inject start commands needed for platform
devices, such as the prueth on the AM65x.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 include/config_distro_bootcmd.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/config_distro_bootcmd.h b/include/config_distro_bootcmd.h
index 2627c2a6a5..6f9b6fda67 100644
--- a/include/config_distro_bootcmd.h
+++ b/include/config_distro_bootcmd.h
@@ -318,6 +318,10 @@
 	BOOT_TARGET_DEVICES_references_VIRTIO_without_CONFIG_CMD_VIRTIO
 #endif
 
+#ifndef BOOTENV_RUN_NET_PLATFORM_START
+#define BOOTENV_RUN_NET_PLATFORM_START
+#endif
+
 #if defined(CONFIG_CMD_DHCP)
 #if defined(CONFIG_EFI_LOADER)
 /* http://www.iana.org/assignments/dhcpv6-parameters/dhcpv6-parameters.xml */
@@ -378,6 +382,7 @@
 		"setenv devtype " #devtypel "; " \
 		BOOTENV_RUN_NET_USB_START \
 		BOOTENV_RUN_PCI_ENUM \
+		BOOTENV_RUN_NET_PLATFORM_START \
 		"if dhcp ${scriptaddr} ${boot_script_dhcp}; then " \
 			"source ${scriptaddr}; " \
 		"fi;" \
@@ -397,6 +402,7 @@
 	"bootcmd_pxe=" \
 		BOOTENV_RUN_NET_USB_START \
 		BOOTENV_RUN_PCI_ENUM \
+		BOOTENV_RUN_NET_PLATFORM_START \
 		"dhcp; " \
 		"if pxe get; then " \
 			"pxe boot; " \
-- 
2.26.2

